core/src/main/codegen/templates/Parser.jj

```diff
@@ -1653,6 +1657,108 @@ SqlNode NamedRoutineCall(
     }
 }
 
+/**
+ * Table parameter of a table function.
+ * The input table with set semantics may be partitioned/ordered on one or more columns.
+ */
+SqlNode TableArg() :
+{
+    final Span s;
+    SqlNode tableRef;
+    SqlNodeList partitionList = SqlNodeList.EMPTY;
+    SqlNodeList orderList = SqlNodeList.EMPTY;
+}
+{
+    { s = span(); }
+    tableRef = ExplicitTable(getPos())
+    [
+        <PARTITION> <BY>
+        partitionList = SimpleIdentifierOrList()
+    ]
+    [
+        orderList = OrderByInTableParam()
+    ]
+    {
+        if (partitionList.isEmpty() && orderList.isEmpty()) {
+          return tableRef;
+        } else {
+          return SqlStdOperatorTable.SET_SEMANTICS_TABLE.createCall(
+              s.pos(), tableRef, partitionList, orderList);
+        }
+    }
+}
+
+SqlNode PartitionedQueryOrQueryOrExpr(ExprContext exprContext) :
+{
+    SqlNode e;
+}
+{
+    e = OrderedQueryOrExpr(exprContext)
+    e = PartitionedByAndOrderBy(e)
+
+    { return e; }
+}
+
+SqlNode PartitionedByAndOrderBy(SqlNode e):
+{
+    final Span s;
+    SqlNodeList partitionList = SqlNodeList.EMPTY;
+    SqlNodeList orderList = SqlNodeList.EMPTY;
+}
```

Comment:
The over window also has partition by order by clause, can that parse block be reused ?

Reply:
The code could not be reused here because partition by clause and order by clause are different  from the partition by clause and order by clause in some other contexts. 
Test cases could cover these cases.
Based on the SQL standard, Order by clause of Polymorphic table function is different from other order by clause.
The list of order keys must be parenthesized if more than one column is listed.
Similarly, PARTITION BY clause with a list of zero or more columns. The list can always be enclosed in parentheses. If there is only one partitioning column, then the parentheses are optional.